clients/js/test/editPool.test.ts

```diff
@@ -15,8 +23,101 @@ import {
   fetchWhitelistV2,
 } from '@tensor-foundation/whitelist';
 import test from 'ava';
-import { fetchPool, getEditPoolInstruction } from '../src/index.js';
-import { ONE_WEEK, createPool, createWhitelistV2 } from './_common.js';
+import {
+  CurveType,
+  PoolType,
+  TENSOR_AMM_ERROR__DELTA_TOO_LARGE,
+  TENSOR_AMM_ERROR__FEES_NOT_ALLOWED,
+  TENSOR_AMM_ERROR__FEES_TOO_HIGH,
+  TENSOR_AMM_ERROR__MAX_TAKER_SELL_COUNT_TOO_SMALL,
+  TENSOR_AMM_ERROR__WRONG_POOL_TYPE,
+  fetchPool,
+  findNftDepositReceiptPda,
+  getCurrentBidPriceSync,
+  getEditPoolInstruction,
+  getSellNftTradePoolInstructionAsync,
+} from '../src/index.js';
+import {
+  MAX_DELTA_BPS,
+  MAX_MM_FEES_BPS,
+  ONE_WEEK,
+  TestAction,
+  createPool,
+  createWhitelistV2,
+  nftPoolConfig,
+  tokenPoolConfig,
+  tradePoolConfig,
+} from './_common.js';
+import { setupLegacyTest, testSell } from './legacy/_common.js';
+
+test('can edit pool with new config', async (t) => {
```

Comment:
2 pretty important tests missing here:

1. it can overwrite makerBroker on edit
2. it can choose to not overwrite brokers - prior makerBroker should stay

Reply:
The contract doesn't currently allow overwriting the `maker_broker` on edit, not sure if it should. @richardwu any reason not to allow this?

Reply:
Seems weird to be able to edit it after you make the order, since it's the originator that should get the credit.

Think we should force them to close/re-open if they really want to change the maker broker.

Reply:
Hmmmmmmm, not sure I agree - this would be very weird, because for marketplace bids we do allow the change of `makerBroker` on bid edits.

I feel like we should be consistent here or that'll cause confusion @richardwu

Reply:
https://github.com/tensor-foundation/marketplace/blob/main/program/src/instructions/bid.rs#L79

there is no check that this is only getting called when initialized (e.g. rentPayer)

Reply:
same with edits on listings (https://github.com/tensor-foundation/marketplace/blob/main/program/src/instructions/edit.rs#L31)

Feels very off to not allow the edit of `makerBroker` on pools then